tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,322 @@
-import type { VaultName } from '@/vaults/types';
 import type { SessionToken } from '@/sessions/types';
 import os from 'os';
 import path from 'path';
 import fs from 'fs';
-import lock from 'fd-lock';
+import { mocked } from 'ts-jest/utils';
+import prompts from 'prompts';
 import Logger, { LogLevel, StreamHandler } from '@matrixai/logger';
-import PolykeyAgent from '@/PolykeyAgent';
+import { Session } from '@/sessions';
 import { sleep } from '@/utils';
-import * as testUtils from './utils';
-
-jest.mock('@/keys/utils', () => ({
-  ...jest.requireActual('@/keys/utils'),
-  generateDeterministicKeyPair:
-    jest.requireActual('@/keys/utils').generateKeyPair,
-}));
+import config from '@/config';
+import * as clientErrors from '@/client/errors';
+import * as testBinUtils from './utils';
 
 /**
- * This test file has been optimised to use only one instance of PolykeyAgent where posible.
- * Setting up the PolykeyAgent has been done in a beforeAll block.
- * Keep this in mind when adding or editing tests.
- * Any side effects need to be undone when the test has completed.
- * Preferably within a `afterEach()` since any cleanup will be skipped inside a failing test.
- *
- * - left over state can cause a test to fail in certain cases.
- * - left over state can cause similar tests to succeed when they should fail.
- * - starting or stopping the agent within tests should be done on a new instance of the polykey agent.
- * - when in doubt test each modified or added test on it's own as well as the whole file.
- * - Looking into adding a way to safely clear each domain's DB information with out breaking modules.
+ * Mock prompts module which is used prompt for password
  */
+jest.mock('prompts');
+const mockedPrompts = mocked(prompts);
 
-describe('Session Token Refreshing', () => {
-  const logger = new Logger('pkStdio Test', LogLevel.WARN, [
+describe('CLI Sessions', () => {
+  const logger = new Logger('status test', LogLevel.WARN, [
     new StreamHandler(),
   ]);
   let dataDir: string;
-  let nodePath: string;
-  let passwordFile: string;
-  let sessionFile: string;
-  let polykeyAgent: PolykeyAgent;
-
-  let tokenBuffer: Buffer;
-  let token: SessionToken;
-  let command: string[];
-  const vaultName = 'TestVault' as VaultName;
-
+  let pkAgentClose;
+  let sessionTokenPath: string;
   beforeAll(async () => {
-    // This handles the expensive setting up of the polykey agent.
+    sessionTokenPath = path.join(global.binAgentDir, config.defaults.tokenBase);
```

Comment:
This doesn't need to be done in `beforeAll`. This can be done statically at the beginning of `describe`.

Reply:
Fixed